### PR TITLE
Migrate non-scenic-based fuchsia tests to cfv2.

### PR DESCRIPTION
**Pull request by chandarrengoog**
Because some unit tests depend on the real Scenic which isn't hermetic, `flutter_runner_tests` and `flutter_runner_scenic_tests` aren't currently compatible with cfv2.
This will change once a FakeScenic is created (https://fxbug.dev/79873).

See https://fxbug.dev/79691.

### Note: This PR depends on [these infra adjustments](https://flutter-review.googlesource.com/c/recipes/+/15380).

CC: @naudzghebre, @keyonghan

### CODE REVIEW

**Comment by chandarrengoog** [comment]
We're using `run-test-suite` for now since `ffx test` doesn't support exclusive test filters: https://fxbug.dev/80338. After `--test-filter` reaches feature parity with gtest filters, I'll make the necessary transitions to use `ffx test`.

**Review by arbreng — APPROVED** [review]

File: tools/fuchsia/fuchsia_archive.gni [hunk 1]
```diff
@@ -174,36 +183,116 @@ template("fuchsia_archive") {
...
       cmx_file = invoker.cmx_file
-    } else {
-      # TODO(fxbug.dev/79691): Default to cml after soft transition.
-      # cml_file =
-      #     rebase_path("//flutter/testing/fuchsia/meta/fuchsia_test.cml")
-      # if (defined(invoker.cml_file)) {
-      #   cml_file = invoker.cml_file
-      # }
-      cmx_file = rebase_path("//flutter/testing/fuchsia/meta/fuchsia_test.cmx")
+
```
[the remark this answers is not in the document]
Comment (chandarrengoog) [reply tools/fuchsia/fuchsia_archive.gni:254]: okay, sounds good.

[the remark this answers is not in the document]
Comment (chandarrengoog) [reply tools/fuchsia/fuchsia_archive.gni:252]: okay, sounds good.

File: testing/fuchsia/meta/test_driver.cml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        },
+    ],
+    use: [
+        {
+            protocol: [
+                "fuchsia.intl.PropertyProvider",
+                "fuchsia.logger.LogSink",
+                "fuchsia.process.Launcher",
+                "fuchsia.tracing.provider.Registry",
+                "fuchsia.vulkan.loader.Loader",
```
[the remark this answers is not in the document]
Comment (chandarrengoog) [reply testing/fuchsia/meta/test_driver.cml]: Surprisingly, it seems like `flutter_runner_tzdata_tests` doesn't try to access this protocol at all.

Instead `shell_tests` fails when this is not present and `ui_tests` tries to access but still passes.

**Pull request merged**